t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -822,6 +861,12 @@ test_expect_success 'sparse-index is not expanded' '
 	ensure_not_expanded reset base -- folder1 &&
 
 	ensure_not_expanded reset --hard update-deep &&
+
+	echo a test change >> sparse-index/README.md &&
```

Comment:
nit: style for tests is to connect the redirect and the file without whitespace:

```
	echo a test change >>sparse-index/README.md &&
```